### PR TITLE
chore: append Azure Stack suffix to custom components images

### PR DESCRIPTION
**Pull request by jadarsie**
**Reason for Change**:

- Append Azure Stack suffix to custom components images
- Update `KUBE_BINARY_URL` default value (missed in #3151)

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [x] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/api/addons_test.go [hunk 1]
```diff
@@ -5171,7 +5171,7 @@ func TestKubeProxyImageSuffix(t *testing.T) {
 			name:       "return empty string if target version is v1.17 or greater",
```
Comment (jackfrancis) [inline pkg/api/addons_test.go:5171]: let's change the name string to reflect what we're testing now

File: parts/k8s/cloud-init/artifacts/cse_install.sh [hunk 2]
```diff
@@ -193,7 +193,7 @@ extractHyperkube() {
   mv "$path/hyperkube" "/usr/local/bin/kubectl-${KUBERNETES_VERSION}"
 }
 extractKubeBinaries() {
-  KUBE_BINARY_URL=${KUBE_BINARY_URL:-"https://dl.k8s.io/v${KUBERNETES_VERSION}/kubernetes-node-linux-amd64.tar.gz"}
+  KUBE_BINARY_URL=${KUBE_BINARY_URL:-"https://kubernetesartifacts.azureedge.net/kubernetes/v${KUBERNETES_VERSION}/binaries/kubernetes-node-linux-amd64.tar.gz"}
```
Comment (jackfrancis) [inline parts/k8s/cloud-init/artifacts/cse_install.sh:196]: @mboersma didn't we do this already in a separate PR?

Comment (mboersma) [reply parts/k8s/cloud-init/artifacts/cse_install.sh:196]: We meant to but apparently didn't: #3151 only changed the URL for this in the VHD.

Comment (jadarsie) [reply parts/k8s/cloud-init/artifacts/cse_install.sh:196]: > Update KUBE_BINARY_URL default value (missed in #3151)

From PR description
